codalab/apps/web/views.py

```diff
@@ -1949,6 +1949,138 @@ def get_success_url(self):
         return reverse('competitions:dumps', kwargs={'competition_pk': dump.competition.pk})
 
 
+class CompetitionWidgetsView(LoginRequiredMixin, DetailView):
+    queryset = models.Competition.objects.all()
+    template_name = 'web/competitions/widgets.html'
+
+    def get_object(self, queryset=None):
```

Comment:
Should this possibly be in `dispatch` instead of `get_object`?

Reply:
It would make sense to add error handling on grabbing the competition here, but it doesn't look like we require that kind of logic

Reply:
Yeah Django should take care of object not found problems, but having this check in `dispatch` seems to make more sense than `get_object`, moving it